modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -188,37 +197,94 @@ func (suite *KeeperTestSuite) TestChanUpgradeTry() {
 				connectionEnd.State = connectiontypes.UNINITIALIZED
 				suite.chainB.GetSimApp().GetIBCKeeper().ConnectionKeeper.SetConnection(suite.chainB.GetContext(), path.EndpointB.ConnectionID, connectionEnd)
 			},
-			false,
+			connectiontypes.ErrInvalidConnectionState,
+		},
+		{
+			"timeout has passed",
+			func() {
+				counterpartyUpgrade.Timeout = types.NewTimeout(clienttypes.NewHeight(0, 1), 0)
+			},
+			types.ErrInvalidUpgrade,
+		},
+		{
+			"initializing handshake fails, proposed connection hops do not exist",
+			func() {
+				proposedUpgrade.Fields.ConnectionHops = []string{ibctesting.InvalidID}
+			},
+			connectiontypes.ErrConnectionNotFound,
+		},
+		{
+			"current upgrade not found even though channel is in INITUPGRADE",
+			func() {
+				// crossing hellos
+				err := path.EndpointB.ChanUpgradeInit()
+				suite.Require().NoError(err)
+
+				suite.chainB.DeleteKey(host.ChannelUpgradeKey(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID))
+			},
+			types.ErrUpgradeNotFound,
+		},
+		{
+			"proposed upgrade fields must be equal to existing upgrade fields in crossing hellos",
+			func() {
+				// crossing hellos
+				err := path.EndpointB.ChanUpgradeInit()
+				suite.Require().NoError(err)
+
+				proposedUpgrade.Fields.ConnectionHops = []string{ibctesting.InvalidID}
+			},
+			types.ErrInvalidUpgrade,
+		},
+		{
+			"startFlushUpgradeHandshake fails due to proof verification failure, counterparty upgrade connection hops are tampered with",
+			func() {
+				counterpartyUpgrade.Fields.ConnectionHops = []string{ibctesting.InvalidID}
+			},
+			commitmenttypes.ErrInvalidProof,
 		},
 		{
-			"invalid connection hops",
+			"startFlushUpgradeHandshake fails due to incompatible upgrades, chainB proposes a new connection hop that does not match counterparty",
+			func() {
+				// reuse existing connection to create a new connection in a non OPEN state
+				connection := path.EndpointB.GetConnection()
+				// ensure counterparty connectionID does not match connectionID set in counterparty proposed upgrade
+				connection.Counterparty.ConnectionId = "connection-50"
+
+				// set proposed connection in state
+				proposedConnectionID := "connection-100"
```

Comment:
side note: I don't think this lint is particularly useful, having string literals works quite well in tests and I think is more readable.

`string "connection-100" has 3 occurrences, make it a constant (goconst)`

Reply:
agree

Reply:
I'm open to turning this setting off?

Reply:
Yea, that sounds good to me. I feel if we define the same const too many times, we will naturally notice

Reply:
Will add a nolint directive for now